owncloudData/src/test/java/com/owncloud/android/data/server/repository/OCServerInfoRepositoryTest.kt

```diff
@@ -0,0 +1,96 @@
+import com.owncloud.android.data.server.datasources.RemoteServerInfoDataSource
+import com.owncloud.android.domain.exceptions.NoConnectionWithServerException
+import com.owncloud.android.domain.server.model.AuthenticationMethod
+import com.owncloud.android.lib.resources.status.OwnCloudVersion
+import com.owncloud.android.testutil.OC_ServerInfo
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class OCServerInfoRepositoryTest {
+    @Rule
+    @JvmField
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    private val remoteServerInfoDataSource = mockk<RemoteServerInfoDataSource>(relaxed = true)
+    private val ocServerInfoRepository: OCServerInfoRepository = OCServerInfoRepository(remoteServerInfoDataSource)
+
+    @Test
+    fun getServerInfoSecureConnection() {
+        every { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) } returns
+                Pair(OwnCloudVersion(OC_ServerInfo.ownCloudVersion), true)
+
+        every { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) } returns
+                AuthenticationMethod.BASIC_HTTP_AUTH
+
+        val serverInfo = ocServerInfoRepository.getServerInfo(OC_ServerInfo.baseUrl)
+        assertEquals(OC_ServerInfo.copy(isSecureConnection = true), serverInfo)
+
+        verify { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) }
+        verify { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) }
+    }
+
+    @Test
+    fun getServerInfoInSecureConnection() {
+        every { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) } returns
+                Pair(OwnCloudVersion(OC_ServerInfo.ownCloudVersion), false)
+
+        every { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) } returns
+                AuthenticationMethod.BASIC_HTTP_AUTH
+
+        val serverInfo = ocServerInfoRepository.getServerInfo(OC_ServerInfo.baseUrl)
+        assertEquals(OC_ServerInfo.copy(isSecureConnection = false), serverInfo)
+
+        verify { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) }
+        verify { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) }
+    }
+
+    @Test
+    fun getServerInfoBearerAuthMethod() {
+        every { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) } returns
+                Pair(OwnCloudVersion(OC_ServerInfo.ownCloudVersion), false)
+
+        every { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) } returns
+                AuthenticationMethod.BEARER_TOKEN
+
+        val serverInfo = ocServerInfoRepository.getServerInfo(OC_ServerInfo.baseUrl)
+        assertEquals(OC_ServerInfo.copy(authenticationMethod = AuthenticationMethod.BEARER_TOKEN), serverInfo)
+
+        verify { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) }
```

Comment:
You are duplicating much code in this class, what about creating some methods like the ones below?

```
prepareServerInfoToBeRetrieved(authenticationMethod: AuthenticationMethod, secureConnection: boolean) {
    every { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) } returns
                Pair(OwnCloudVersion(OC_ServerInfo.ownCloudVersion), secureConnection)
    every { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) } returns
                authenticationMethod
}
```

```
verifyServerInfoIsRetrievedFromRemoteDataSource {
    verify { remoteServerInfoDataSource.getRemoteStatus(OC_ServerInfo.baseUrl) }
    verify { remoteServerInfoDataSource.getAuthenticationMethod(OC_ServerInfo.baseUrl) }
}
```